fastlane/lib/lanes/localization.rb

```diff
@@ -8,69 +53,91 @@
 
-        utf16_to_utf8(
-          source: utf16_strings,
-          destination: utf8_strings
-        )
-      end
+    SOURCES_TO_LOCALIZE.each do |source|
+      next if source.gp_project_url.nil?
+
+      ios_download_strings_files_from_glotpress(
+        project_url: source.gp_project_url,
+        locales: GLOTPRESS_TO_LPROJ_APP_LOCALE_CODES,
+        download_dir: source.localizations_root
+      )
+
+      next if skip_commit
+
+      git_add(path: source.localizations_root)
+      paths_to_commit << source.localizations_root
     end
-  
-    lane :generate_strings_file_sdk do |options|
+
+    next if skip_commit
+
+    git_commit(
+      path: paths_to_commit,
+      message: 'Update localizations',
+      allow_nothing_to_commit: true
+    )
+  end
+
+  # Generates the `.strings` files for the base language by parsing source code (using `genstring`).
+  #
+  lane :generate_strings do |skip_commit: false|
+    paths_to_commit = []
+
+    SOURCES_TO_LOCALIZE.each do |source|
       Dir.mktmpdir do |tempdir|
-        demo_en_lproj = File.join('Sources', 'GravatarUI', 'Resources', 'en.lproj')
         ios_generate_strings_file_from_code(
-          paths: [
-            File.join('Sources', 'GravatarUI')
-          ],
+          paths: source.source_paths,
           output_dir: tempdir
         )
-  
+
         utf16_strings = File.join(tempdir, 'Localizable.strings')
-        utf8_strings = File.join("..", demo_en_lproj, 'Localizable.strings')
+        utf8_strings = File.join('..', source.base_localization_strings)
 
         utf16_to_utf8(
           source: utf16_strings,
           destination: utf8_strings
         )
       end
+
+      next if skip_commit
+
+      git_add(path: source.localizations_root)
+      paths_to_commit << source.localizations_root
     end
-  
-    private_lane :utf16_to_utf8 do |options|
-      next unless options[:source]
-      next unless options[:destination]      
-  
-      source = options[:source]
-      destination = options[:destination]
-  
-      next unless File.exist?(source)
-      File.open(source, "rb:UTF-16") do |in_file|
-        utf16_content = in_file.read
-        utf8_content = utf16_content.encode("UTF-8")
-
-        File.open(destination, "w:UTF-8") do |out_file|
-          out_file.write(utf8_content)
-        end
+
+    next if skip_commit
+
+    git_commit(
+      path: paths_to_commit,
+      message: 'Update strings in base locale',
+      allow_nothing_to_commit: true
+    )
+  end
+
+  private_lane :utf16_to_utf8 do |options|
+    next unless options[:source]
+    next unless options[:destination]
+
+    source = options[:source]
+    destination = options[:destination]
+
+    next unless File.exist?(source)
+
+    File.open(source, 'rb:UTF-16') do |in_file|
+      utf16_content = in_file.read
+      utf8_content = utf16_content.encode('UTF-8')
+
+      File.open(destination, 'w:UTF-8') do |out_file|
+        out_file.write(utf8_content)
       end
     end
   end
```

Comment:
In other words, writing this as a regular Ruby function should look like this, iinm (writing from GitHub, Ruby syntax not validated in a REPL, so I'll let you validate it works):

```suggestion
  def utf16_to_utf8(source:, destination:)
    return unless File.exist?(source)

    utf16_content = File.read(source, 'rb:UTF-16') do |in_file|
    utf8_content = utf16_content.encode('UTF-8')
    File.write(destination, utf8_content, 'w:UTF-8')
  end
```

Reply:
Nice.  I'm not able to use the `File.read(source, 'rb:UTF-16')` (or `File.read(source, encoding: 'UTF-16')`) syntaxes because these files are binary (unless there's a way to use them that I'm not aware of).

Reply:
What about `File.read(source, mode: 'rb:UTF-16')`, or `File.read(source, binmode: true, encoding: 'UTF-16')`? 🤔

Reply:
phew, that reads so much better.  Thanks.  I was tired but also unwilling to stop on Friday evening. 😁